lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2980,14 +2980,20 @@ LogicalResult FIRRTLLowering::visitDecl(MemOp op) {
   // Memories return multiple structs, one for each port, which means we
   // have two layers of type to split apart.
   for (size_t i = 0, e = op.getNumResults(); i != e; ++i) {
-    auto addInput = [&](StringRef field, size_t width,
-                        StringRef field2 = "") -> Value {
-      auto portType =
-          IntegerType::get(op.getContext(), std::max<size_t>(1, width));
-
-      Value backedge = createBackedge(builder.getLoc(), portType);
+    auto addInput = [&](StringRef field, size_t width) -> Value {
       auto accesses = getAllFieldAccesses(op.getResult(i), field);
 
+      // If the memory is 0-width, do not materialize any connections to it.
+      // However, `seq.firmem` now requires a 1-bit input, so materialize
+      // a dummy x value to provide it with.
+      Value backedge, portValue;
+      if (width == 0) {
+        portValue = getOrCreateXConstant(1);
+      } else {
+        auto portType = IntegerType::get(op.getContext(), width);
+        backedge = portValue = createBackedge(builder.getLoc(), portType);
+      }
```

Comment:
It wasn't obvious to me that `setLowering` could accept a null `backedge`. However, this is fine after checking that function.